src/AbstractAppliance.js

```diff
@@ -0,0 +1,119 @@
+import assert from 'assert'
+import EventEmitter from 'events'
+import {
+	AbstractInstantiationError,
+	NotImplementedError,
+} from './errors'
+import PayloadBuffer from './PayloadBuffer'
+import { isPayloadInstance } from './utils/payload'
+
+/**
+ * The Abstract Appliance defines the shape of an Appliance and implements
+ * a few universal pieces of functionality, such as:
+ *
+ * - Event emission
+ * - Payload validation process flow
+ * - Appliance setting resolution
+ *
+ * AbstractAppliance methods prefixed with `_` require implementation.
+ * Other methods, aside from the constructor, should not be overridden.
+ *
+ * For more information about the TV Kitchen architecture visit:
+ * https://github.com/tvkitchen/tv-kitchen/blob/master/docs/ARCHITECTURE.md
+ */
+class AbstractAppliance {
+	settings = {}
+
+	emitter = new EventEmitter()
+
+	payloadBuffer = new PayloadBuffer()
+
+	constructor(overrideSettings = {}) {
+		if (this.constructor === AbstractAppliance) {
+			throw new AbstractInstantiationError('AbstractAppliance')
+		}
+		Object.assign(
+			this.settings,
+			overrideSettings,
+		)
+	}
+
+	/**
+	 *******************************************
+	 ************ ABSTRACT METHODS *************
+	 *******************************************
+	 */
+
+	/**
+	 * Getter for the list of data types accepted by the appliance.
+	 *
+	 * OVERRIDE WHEN EXTENDING
+	 *
+	 * @return {Array[String]} The list of data types accepted by the appliance.
+	 */
+	getInputTypes = () => {
+		throw new NotImplementedError('getInputTypes')
+	}
+
+	/**
+	 * Getter for the list of data types produced by the appliance.
+	 *
+	 * OVERRIDE WHEN EXTENDING
+	 *
+	 * @return {Array[String]} The list of data types produced by the appliance.
+	 */
+	getOutputTypes = () => {
+		throw new NotImplementedError('getOutputTypes')
+	}
+
+	/**
+	 * Asserts that a given payload is actually a payload.
+   *
+   * OVERRIDE WHEN EXTENDING
+   *
+	 * @param  {Payload} payload The payload that we want to validate.
+	 * @return {Boolean}         The result of the validation check.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isValidPayload = async (payload) => {
```

Comment:
Why is this async? I don't doubt that you know it will need to be, but I can't 🤔 precisely why.

Reply:
My logic was "I can imagine a universe where the logic to check that a payload is valid requires the use of a third party service and therefore would need to be async".

Making a method async is more flexible than making a method sync, so I leaned towards it here (especially a class with the level of ecosystem-wide impact as this)

Reply:
(let me know if that logic feels sound to you as well btw)